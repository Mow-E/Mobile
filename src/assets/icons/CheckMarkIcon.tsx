import * as React from 'react';
import Svg, {SvgProps, Line} from 'react-native-svg';
import IconProps from './IconProps';
import colors from '../../styles/colors';

function SvgComponent(props: SvgProps): JSX.Element {
  return (
    <Svg width={24} height={24} viewBox="0 0 27 23" fill="none" {...props}>
      <Line
        x1={0.573462}
        y1={15.1808}
        x2={10.5735}
        y2={22.1808}
        stroke={props.stroke}
        strokeWidth={2}
      />
      <Line
        x1={9.22276}
        y1={22.3708}
        x2={26.2228}
        y2={1.3708}
        stroke={props.stroke}
        strokeWidth={2}
      />
    </Svg>
  );
}

/**
 * An icon for a check mark.
 */
function CheckMarkIcon({size = 24, colored = false}: IconProps): JSX.Element {
  const strokeColor =
    typeof colored === 'boolean' ? colors.gray['950'] : colored;

  return <SvgComponent height={size} width={size} stroke={strokeColor} />;
}

export default CheckMarkIcon;
