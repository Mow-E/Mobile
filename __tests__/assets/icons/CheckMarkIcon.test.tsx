import 'react-native';
import React from 'react';
import CheckMarkIcon from '../../../src/assets/icons/CheckMarkIcon';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders default correctly', () => {
  const tree = renderer.create(<CheckMarkIcon />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders colored correctly', () => {
  const tree = renderer.create(<CheckMarkIcon colored />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders uncolored correctly', () => {
  const tree = renderer.create(<CheckMarkIcon colored={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders in dark mode correctly', () => {
  const tree = renderer.create(<CheckMarkIcon darkModeInverted />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders in different sizes', () => {
  const treeSmall = renderer.create(<CheckMarkIcon size={24} />).toJSON();
  const treeLarge = renderer.create(<CheckMarkIcon size={48} />).toJSON();

  expect(treeSmall).not.toEqual(treeLarge);
});
