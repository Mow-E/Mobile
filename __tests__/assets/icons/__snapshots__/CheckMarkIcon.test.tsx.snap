// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders colored correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={23}
  vbWidth={27}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="2"
      x1="0.573462"
      x2="10.5735"
      y1="15.1808"
      y2="22.1808"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="2"
      x1="9.22276"
      x2="26.2228"
      y1="22.3708"
      y2="1.3708"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders default correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={23}
  vbWidth={27}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="2"
      x1="0.573462"
      x2="10.5735"
      y1="15.1808"
      y2="22.1808"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="2"
      x1="9.22276"
      x2="26.2228"
      y1="22.3708"
      y2="1.3708"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders in dark mode correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={23}
  vbWidth={27}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="2"
      x1="0.573462"
      x2="10.5735"
      y1="15.1808"
      y2="22.1808"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="2"
      x1="9.22276"
      x2="26.2228"
      y1="22.3708"
      y2="1.3708"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders uncolored correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={23}
  vbWidth={27}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="2"
      x1="0.573462"
      x2="10.5735"
      y1="15.1808"
      y2="22.1808"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="2"
      x1="9.22276"
      x2="26.2228"
      y1="22.3708"
      y2="1.3708"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
